src/couldBeSharedComponents/LayoutPanels/LayoutPanels.test.tsx

```diff
@@ -45,3 +46,123 @@ describe("Layout Panels default", async () => {
   });
 });
+describe("Layout Panels using a subpanel", () => {
+  it("Throws an error if no external subpanel state is provided", () => {
+    expect(() =>
+      render(
+        <LayoutPanels
+          leftPanelContent={<>left panel content</>}
+          rightPanelContent={<>right panel content</>}
+          subpanelContent={<>subpanel content</>}
+        >
+          main content
+        </LayoutPanels>,
+      ),
+    ).toThrow();
+    expect(() =>
+      render(
+        <LayoutPanels
+          leftPanelContent={<>left panel content</>}
+          rightPanelContent={<>right panel content</>}
+          subpanelContent={<>subpanel content</>}
+          setIsSubpanelOpen={() => {}} // setter but missing state
+        >
+          main content
+        </LayoutPanels>,
+      ),
+    ).toThrow();
+    expect(() =>
+      render(
+        <LayoutPanels
+          leftPanelContent={<>left panel content</>}
+          rightPanelContent={<>right panel content</>}
+          subpanelContent={<>subpanel content</>}
+          isSubpanelOpen={true} // state but missing setter
+        >
+          main content
+        </LayoutPanels>,
+      ),
+    ).toThrow();
+    expect(() =>
+      render(
+        <LayoutPanels
+          leftPanelContent={<>left panel content</>}
+          rightPanelContent={<>right panel content</>}
+          subpanelContent={<>subpanel content</>}
+          isSubpanelOpen={true}
+          setIsSubpanelOpen={() => {}}
+        >
+          main content
+        </LayoutPanels>,
+      ),
+    ).not.toThrow();
+  });
+  it("Opens and closes the subpanel as expected", () => {
+    const ControlledLayout = () => {
+      const [isSubpanelOpen, setIsSubpanelOpen] = useState(true);
+
+      const toggleSubpanel = () => {
+        setIsSubpanelOpen((previous) => !previous);
+      };
+
+      return (
+        <>
+          <LayoutPanels
+            leftPanelContent={<>left panel content</>}
+            rightPanelContent={<>right panel content</>}
+            subpanelContent={<>subpanel content</>}
+            isSubpanelOpen={isSubpanelOpen}
+            setIsSubpanelOpen={setIsSubpanelOpen}
+          >
+            main content
+          </LayoutPanels>
+          <button onClick={toggleSubpanel}>toggle subpanel</button>
+        </>
+      );
+    };
+    render(<ControlledLayout />);
+
+    const subpanelToggleButton = screen.getByRole("button", {
+      name: "toggle subpanel",
+    });
+    expect(screen.getByText("subpanel content")).toBeInTheDocument();
+    fireEvent.click(subpanelToggleButton);
+    expect(screen.queryByText("subpanel content")).not.toBeInTheDocument();
+    fireEvent.click(subpanelToggleButton);
+    expect(screen.getByText("subpanel content")).toBeInTheDocument();
+  });
+  it("Automatically closes the subpanel when the left panel is closed. When the left panel is opened, the subpanel is not opened automatically.", () => {
```

Comment:
I used to be a real purist about not combining too much in one test, but on MERMAID we ran into performance issues that were solved by combining multiple related tests into one. That uses Jest not Vitest, and MERMAID has a huge dom compared to this tiny layout being rendered per test, but I'm experimenting with a default of keeping stuff together where its closely related. Like in this test, even though it makes the `it` grammar weird. 

For performance, and also to reuse the test setup where it doesnt cause test isolation issues.